### ISSUE TITLE
Interpreting Captum Regression Results with Language Models

### ISSUE THREAD

**Issue by RylanSchaeffer** [pytorch/captum#786]
I adapted the Captum regression tutorial (https://captum.ai/tutorials/House_Prices_Regression_Interpret) to fine tuning a transformer language model (DistilBert, to be precise) on a regression task, and I need help understanding what red/green mean in a regression language modeling context.

![image](https://user-images.githubusercontent.com/8942987/136266035-98ceabba-b2ce-4c27-82a3-828c4fef6253.png)

Does green mean lowering the mean-squared error? If so, shouldn't I need to provide the target in the `.attribute()` function call? If the target isn't provided, what do green/red mean?

```
        attributions, delta = lig.attribute(
            inputs=input_ids_and_attention_masks['input_ids'],
            baselines=reference_indices,
            additional_forward_args=(input_ids_and_attention_masks['attention_mask'],),
            # target=target,
            n_steps=50,
            return_convergence_delta=True)

```

**Comment by RylanSchaeffer** [comment]
Something looks wrong because certain words (e.g. explorer) seem useful for certain data points (e.g. the first example above) but useless for other data points (e.g. all other examples above), even though that word (explorer) is equally present for all data points.

**Comment by NarineK** [comment]
@RylanSchaeffer, the color coding has nothing to do with the accuracy of the model. It shows the importance score of input features for the output that your predictor returned. In your case it can be that the word `explorer` contributed to the prediction in one case and didn't contribute in the other case.
What does the model return ? What does predicted score indicate ? Is this how you compute attributions ?
```
        attributions, delta = lig.attribute(
            inputs=input_ids_and_attention_masks['input_ids'],
            baselines=reference_indices,
            additional_forward_args=(input_ids_and_attention_masks['attention_mask'],),
            # target=target,
            n_steps=50,
            return_convergence_delta=True)
```
What value do you get for delta ?

**Comment by RylanSchaeffer** [comment]
> It shows the importance score of input features for the output that your predictor returned.

Could you explain in more detail what green and red mean? Red in the legend is labeled "Negative" - does that then mean that red words are irrelevant to the prediction?

![image](https://user-images.githubusercontent.com/8942987/136301983-19c31aa9-e31e-44b3-bb09-d1c7d7422c97.png)

**Comment by NarineK** [comment]
red means that they are unimportant for that specific prediction. They are negatively correlated with the predicted scrore.

**Comment by NarineK** [comment]
red means that they are unimportant for that specific prediction. They are negatively correlated with the predicted score.

**Comment by RylanSchaeffer** [comment]
What does "important" mean in the context of regression?

**Comment by RylanSchaeffer** [comment]
@NarineK I found a highly relevant ICLR 2018 paper (https://arxiv.org/pdf/1711.06104.pdf) that suggests attribution colors are meant to indicate the effect on the output activation:

 "When the attributions of all input features are arranged together to have the same shape of the input sample we talk about attribution maps (Figures 1-2), which are usually displayed as heatmaps where red color indicates features that contribute positively to the activation of the target output, and blue color indicates features that have a suppressing effect on it."
